core/dbt/include/global_project/macros/materializations/models/table/create_table_as.sql

```diff
@@ -25,11 +25,26 @@
 
   create {% if temporary: -%}temporary{%- endif %} table
     {{ relation.include(database=(not temporary), schema=(not temporary)) }}
-    {% if config.get('contract', False) %}
-      {{ get_assert_columns_equivalent(sql) }}
-      {{ get_columns_spec_ddl() }}
-    {% endif %}
+  {% if config.get('contract', False) %}
+    {{ get_assert_columns_equivalent(sql) }}
+    {{ get_columns_spec_ddl() }}
+    {%- set sql = get_select_subquery(sql) %}
+  {% endif %}
   as (
     {{ sql }}
   );
 {%- endmacro %}
+
+{% macro get_select_subquery(sql) -%}
+  {{ return(adapter.dispatch('get_select_subquery', 'dbt')(sql)) }}
+{% endmacro %}
+
+{% macro default__get_select_subquery(sql) -%}
+    select
+    {% for column in model['columns'] %}
+      {{ column }}{{ ", " if not loop.last }}
+    {%- endfor %}
```

Comment:
to match other patterns 
```suggestion
    {%- for column in model['columns'] -%}
      {{ column }}{{ ", " if not loop.last }}
    {%- endfor -%}
```

Reply:
Will add some hyphens.

Reply:
Hyphens on these lines breaks the generated sql.